client/src/state/secure.rs

```diff
@@ -11,34 +11,29 @@ use engine::{
     store::Cache,
     vault::{ClientId, DbView, RecordId, VaultId},
 };
-use std::{collections::HashSet, time::Duration};
+use std::time::Duration;
 
 /// Cache type definition
 pub type Store = Cache<Vec<u8>, Vec<u8>>;
 
 pub struct SecureClient {
     // A keystore
-    pub(crate) keystore: KeyStore<internals::Provider>,
+    pub(crate) keystore: KeyStore,
```

Comment:
might be a breaking change for the snapshot data.  Would recommend checking to see if the bincode representation of the `KeyStore` is the same as `KeyStore<P>`.

Reply:
I don't think it changes anything for the snapshot data. 
The messages that gather the Information from the `SecureClient` (`GetData`) and send it to the `SnapshotActoor` (`FillSnapshot`) do not include the `Keystore`, but instead only the `Key` and the `DbView`:
```
pub struct GetData {}
impl Message for GetData {
    type Result = Result<
        Box<(
            HashMap<VaultId, Key<internals::Provider>>,
            DbView<internals::Provider>,
            Store,
        )>,
        anyhow::Error,
    >;
}

pub struct FillSnapshot {
    pub data: Box<(HashMap<VaultId, Key<Provider>>, DbView<Provider>, Store)>,
    pub id: ClientId,
}
```
Since they already used the concrete `internals::Provider` and were not changed in this PR, I don't think it changed anything for the snapshot.
Please correct me in case that I am missing something. I agree that this is something we should be very careful with.

Reply:
Correct.  It should be ok since the state uses the provider directly in there and also doesn't use the keystore.